src/test/java/org/mule/tools/apikit/AbstractScaffolderTestCase.java

```diff
@@ -40,6 +48,63 @@ public void after() {
     System.clearProperty(TestUtils.PARSER_V2_PROPERTY);
   }
 
+  protected static void verifySuccessfulScaffolding(ScaffoldingResult result, String... expectedArrayPath)
+      throws IOException, SAXException {
+    List<String> expectedPaths = Arrays.asList(expectedArrayPath);
+    assertTrue(result.isSuccess());
+    assertEquals(expectedPaths.size(), result.getGeneratedConfigs().size());
+    for (String expectedPath : expectedArrayPath) {
+      String expected = APIKitTools.readContents(getResourceAsStream(expectedPath));
+      String configFileName = extractName(expectedPath);
+      String generated = retrieveGeneratedFile(result, configFileName);
+      XMLUnit.setIgnoreWhitespace(true);
+      Diff diff = XMLUnit.compareXML(expected, generated);
+      assertTrue(diff.identical());
+    }
+  }
+
+  protected static String retrieveGeneratedFile(ScaffoldingResult secondScaffoldingResult, String fileName)
+      throws IOException {
+    InputStream api = secondScaffoldingResult.getGeneratedConfigs().stream().filter(config -> config.getName().contains(fileName))
+        .findFirst().get().getContent();
```

Comment:
You can do something like this to avoid NPE and provide more details for the failing test
```.findFirst().orElseThrow(() -> new RuntimeException("Unable to find generated file")).getContent();```